app/assets/javascripts/components/dropdown.js

```diff
@@ -0,0 +1,44 @@
+function Dropdown ($module) {
+  if (!$module) $module = document
+  this.$module = $module
+  this.$title = $module.querySelector('.app-c-dropdown__title')
+  this.$container = $module.querySelector('.app-c-dropdown__container')
+  this.$buttons = $module.querySelectorAll('.app-c-dropdown__button')
+}
+
+Dropdown.prototype.handleBlur = function (event) {
```

Comment:
I'm guessing this is partly here to cope with the fact that we override the button click events in the inline-image-module? It took a little while to understand the `relatedTarget` stuff, although I'm still not sure why we can't just do `this.closeDropdown()` on blur, as it checks the `open` state. Can you explain a bit?

Reply:
Blur can happen for both moving to an element inside or outside the component. With `relatedTarget` we try to capture where the user goes to after opening the dropdown. If he goes inside, we don't want to close the dropdown.